### ISSUE TITLE
How do you hide the TitleBar of the Ribbon

### ISSUE THREAD

**Issue by rjusher** [fluentribbon/Fluent.Ribbon#73]
How do you make the title bar in the Ribbon to be hidden, better if its visibility could be changed to collapsed.

**Comment by batzen** [comment]
What do you need this feature for?

**Comment by rjusher** [comment]
I was working with the another ribbon component which has this capability, but recently was asked to remove all the dependencies on this components library. So the component I was asked to remove was used to present in a ribbon way the top menu, but was kept minimized so it allowed lot of space for not navigation work. When changed to this component i couldn't find a way to recover the vertical space, currently I am keeping the ribbon minimized and added the following in the ribbon styles:

var titleBarStyle = new Style(typeof (RibbonTitleBar));
titleBarStyle.Setters.Add(new Setter(UIElement.VisibilityProperty, Visibility.Hidden));
item.Resources.Add("RibbonTitleBarStyle", titleBarStyle);
var quickAccessStyle = new Style(typeof (QuickAccessToolBar));
quickAccessStyle.Setters.Add(new Setter(UIElement.VisibilityProperty, Visibility.Hidden));
item.Resources.Add("QuickAccessToolbarStyle ", quickAccessStyle);

To hide the Title, because the form title doesn't really represent the title it must show, and I believe the component is bound to the containing form title.

Long story short: I use the ribbon as a rich menu, but want the most of the vertical space I can provide my users.

Thanks

**Comment by pekspro** [comment]
I have made a fork of the Fluent Ribbon, where I have added an option to use the default window frame. You find it here:

https://github.com/pekspro/Fluent.Ribbon

If you add UseDefaultWindowsChrome="True" as an attribute to the windows the default frame will be used. If you also add WindowStyle="None" the title bar will be removed.

I’m not sure that is exactly what you are looking for. Also note than when you do this some other elements that are in the title bar will be removed (like contextual ribbons etc).

**Comment by batzen** [comment]
If you don't want to use our window class, why don't you just use your own or the default WPF window class?

**Comment by pekspro** [comment]
Hmm… that’s a really good point :-). However, I’m not sure that’s enough. If I change the window in the test project to the default wpf window class I still get the Fluent Ribbon title bar inside that window.

**Comment by batzen** [comment]
Strange as it's working with the Mah-Metro window.

**Comment by batzen** [comment]
Fluent:RibbonProperties.TitleBarHeight="0"
and you are done hiding the titlebar. ;-)

**Comment by rjusher** [comment]
Thanks @pekspro I'll further watch your fork for the feature you mention, but for now the solution @batzen gave, solves my problem.

**Comment by pekspro** [comment]
Interesting, maybe I could get rid of my fork :-). The only potential problem I see right now is that you still could use tab to get into the quick access toolbar. But I haven’t tried it, I will be away for a couple of days and will check this out when I’m back.

**Comment by pekspro** [comment]
As suggested above a changed my application to the use original Windows class instead of FluentWindow. I also added Fluent:RibbonProperties.TitleBarHeight="0" to remove the title bar and added IsQuickAccessToolBarVisible="False" to hide the quick access toolbar (which still was accessible with the tab key).

And it works perfectly! I no longer need my fork – Thanks! Yeah! :-)

**Comment by cosijuan** [comment]
In the Load event of the window, add the next line:

((System.Windows.UIElement)((System.Windows.FrameworkElement)(this.RibbonMain.Template.FindName("PART_TitleHost", this.RibbonMain) as ContentPresenter).Parent).Parent).Visibility = Visibility.Collapsed;
